runtime/vm/linearswalk.c

```diff
@@ -427,23 +433,34 @@ lswDescribeSlot(J9VMThread * vmThread, J9StackWalkState * walkState, J9SWSlot *
 		case LSW_TYPE_INDIRECT_O_SLOT:
 		case LSW_TYPE_O_SLOT:
 			if (slot->data) {
-			
+				char *newPtr = NULL;
 				if (slot->type == LSW_TYPE_INDIRECT_O_SLOT) {
+					int len = 0;
 					object = (J9Object *) *((UDATA *) (slot->data & ~1));
-					dataPtr += sprintf(dataPtr, "%p -> ", object);
+					len = snprintf(dataPtr, dataSize, "%p -> ", object);
+					if ((0 < len) && ((size_t)len <= dataSize)) {
+						dataPtr += len;
+						dataSize -= len;
+					} else {
+						dataSize = 0;
+					}
 				} else {
 					object = (J9Object *) slot->data;
 				}
 				
 				object = (J9Object *) object;
 				class = J9OBJECT_CLAZZ(vmThread, object);
 				
-				dataPtr = lswDescribeObject(vmThread, dataPtr, &class, object);
-
-				className = J9ROMCLASS_CLASSNAME(class->romClass);
-				copySize = (J9UTF8_LENGTH(className) <= dataSize) ? J9UTF8_LENGTH(className) : LSW_STRING_MAX - 1;
-				memcpy(dataPtr, J9UTF8_DATA(className), copySize);
-				dataPtr[copySize] = 0;
+				newPtr = lswDescribeObject(vmThread, dataPtr, dataSize, &class, object);
+				dataSize -= newPtr - dataPtr;
+				dataPtr = newPtr;
+
+				if (dataSize > 0) {
+					className = J9ROMCLASS_CLASSNAME(class->romClass);
+					copySize = (J9UTF8_LENGTH(className) < dataSize) ? J9UTF8_LENGTH(className) : (0 < dataSize ? (dataSize - 1) : 0);
```

Comment:
`(0 < dataSize ? (dataSize - 1) : 0)` can be replaced by `(dataSize - 1)`.

Reply:
Updated.